### ISSUE TITLE
Need to support Apple silicon

### ISSUE THREAD

**Issue by hickey** [particledecay/asdf-lab#10]
**Is your feature request related to a problem? Please describe.**

Yes. When attempting to install `lab` for machines with Apple silicon the architecture name is not included in the URL. 

Here is the `uname` outputs for machines with Apple silicon:

```shell
❯ uname
Darwin
❯ uname -m
arm64
```

The download URL that is being attempted is:

`https://github.com/zaquestion/lab/releases/download/v0.25.1/lab_0.25.1_darwin_.tar.gz`

Whereas the correct download URL is: 

`https://github.com/zaquestion/lab/releases/download/v0.25.1/lab_0.25.1_darwin_arm64.tar.gz`

**Describe the solution you'd like**

The plugin to support the `arm64` architecture.

**Comment by particledecay** [comment]
Thanks for the report, I'll take a look tonight!

**Comment by particledecay** [comment]
@hickey would you mind pasting the output of `uname -a` on the system you're having this problem with?

**Comment by hickey** [comment]
```shell
❯ uname -a
Darwin V57CXMQMNW 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:10:42 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6000 arm64
```

**Comment by particledecay** [comment]
Are you on an M1 Mac Mini? Or have Rosetta enabled? I saw a bug listed somewhere that the arch returned isn't accurate in some related situations... but in my (admittedly limited) testing I can't seem to reproduce what you're seeing.

Any chance you could give more details? Like are you running this from a VSCode terminal, what shell, tmux environment, etc?

**Comment by hickey** [comment]
MacBook Pro 2021. Model identifier MacBookPro18,3

Running in a terminal. iTerm2 specifically. But that would have no effect on the `uname` output. `zsh` instead of `bash`, but again that would have no effect on `uname`.

**Comment by particledecay** [comment]
You'd think so, but I found a few things that suggested otherwise, like [this](https://stackoverflow.com/questions/76024834/why-does-uname-return-different-current-hardware-platform-in-different-terminals). I'm just trying to figure out what else might be going on.

The logic to retrieve the architecture is [here](https://github.com/particledecay/asdf-lab/blob/main/lib/utils.bash#L85-L96), which is simple and functions the same as other plugins I maintain (which works on all my machines). It's then added as part of the download URL [here](https://github.com/particledecay/asdf-lab/blob/main/lib/utils.bash#L42).

I have a system with a T6000 variant M1 that I can maybe test later tonight and see what's up, but otherwise I'm at a loss.

**Comment by hickey** [comment]
Yes, that is true if you are running under an emulated environment like Rosetta. It would be the same if I were giving you the `uname` output for a shell running inside a Docker container. Of course it is not going to report the correct processor and OS labels. That is why I was giving you the true output from `uname`. 

Now I did find the problem. A couple of years ago I setup `asdf` which at the time the install would have one checkout their `v.0.8.0` tag. I keep a number of scripts for setting up the environments between multiple OSes and a script was created for installing `asdf` and another for the `asdf` plugins (along with loading up all the tools and required versions). Well the `asdf` script has never been updated and why would it as `asdf` keeps working like a champ. 

I was looking at your code again and realized that the architecture should be returned by `get_arch()` but that is not what seemed to be happening. I was going to manually go modify the code in the plugin directory so I could get some diagnostics while trying to install `lab`. When I got to the plugin dir what was in the directory was not what was showing in the repo. Specifically I was missing the `lib` dir. I am not sure how or why but it appears that the old `asdf` version was only getting the first commit in your repo. (I just looked and it seems it was getting the `master` branch rather than the `main` branch) 

Once I did a `asdf update` to get the `v0.14.0` version (I also removed and re-added `asdf-lab`) I could then install `lab` or Apple silicon. 

Sorry to have you spend time on a needless issue.
